chains/orchestrator-relays/runtime/dancelight/src/tests/container_token_transfers.rs

```diff
@@ -848,3 +848,331 @@ fn native_container_can_process_message_returns_true_for_valid_message() {
+                event_log: Log {
+                    address:
+                        <Runtime as snowbridge_pallet_inbound_queue::Config>::GatewayAddress::get(),
+                    topics: event
+                        .encode_topics()
+                        .into_iter()
+                        .map(|w| H256::from(w.0 .0))
+                        .collect(),
+                    data: event.encode_data(),
+                },
+                proof: mock_snowbridge_message_proof(),
+            };
+
+            assert_ok!(EthereumInboundQueue::submit(relayer, message));
+
+            let sent = System::events().iter().any(|r| {
+                matches!(
+                    r.event,
+                    RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. })
+                )
+            });
+            assert!(sent, "XCM Sent event should be emitted!");
+        });
+}
+
+#[test]
+fn receive_container_foreign_tokens_from_eth_works_for_foreign_account_id_32() {
+    ExtBuilder::default()
+        .with_balances(vec![(AccountId::from(ALICE), 100_000 * UNIT)])
+        .build()
+        .execute_with(|| {
+            sp_tracing::try_init_simple();
+
+            let relayer = RuntimeOrigin::signed(AccountId::from(ALICE));
+            let para_id: ParaId = 2000u32.into();
+            let container_para_id = 2001u32;
+
+            assert_ok!(XcmPallet::force_default_xcm_version(
+                root_origin(),
+                Some(5u32)
+            ));
+
+            assert_ok!(Paras::force_set_current_head(
+                root_origin(),
+                container_para_id.into(),
+                HeadData::from(vec![1u8, 2u8, 3u8])
+            ));
+
+            let channel_id = ChannelId::new([1u8; 32]);
+            let agent_id = AgentId::from_low_u64_be(42);
+
+            assert_ok!(EthereumTokenTransfers::set_token_transfer_channel(
+                root_origin(),
+                channel_id,
+                agent_id,
+                para_id
+            ));
+
+            let token_addr = H160::repeat_byte(0x11);
+            let token_location = Location {
+                parents: 1,
+                interior: X2([
+                    GlobalConsensus(EthereumNetwork::get()),
+                    AccountKey20 {
+                        network: Some(EthereumNetwork::get()),
+                        key: token_addr.into(),
+                    },
+                ]
+                .into()),
+            };
+
+            let asset_id = 42u16;
+
+            assert_ok!(
+                pallet_foreign_asset_creator::Pallet::<Runtime>::create_foreign_asset(
+                    root_origin(),
+                    token_location.clone(),
+                    asset_id,
+                    AccountId::from(ALICE),
+                    true,
+                    1
+                )
+            );
+
+            let beneficiary = [5u8; 32];
+
+            let amount_to_transfer = 100_000_000;
```

Comment:
how does this work? how are we paying for fees in the container? with the ethereum token? and if so, the container_fee is less than the `amount_to_transfer`, how is that possible?